sqrl-server/sqrl-server-vertx-base/src/main/java/com/datasqrl/graphql/JdbcClientsConfig.java

```diff
@@ -17,94 +17,80 @@
-    snowflakeUrl.ifPresent(url -> clients.put(DatabaseType.SNOWFLAKE, createSnowflakeClient(url)));
-    return clients;
-  }
+    var builder =
+        ImmutableMap.<DatabaseType, SqlClient>builder()
+            .put(DatabaseType.POSTGRES, initPostgresSqlClient());
 
-  @SneakyThrows
-  private static Optional<String> readSnowflakeUrl() {
-    File snowflakeConfig = new File("snowflake-config.json");
-    Map map = null;
-    if (snowflakeConfig.exists()) {
-      map = HttpServerVerticle.getObjectMapper().readValue(snowflakeConfig, Map.class);
-      if (map.isEmpty()) {
-        return Optional.empty();
-      }
-    } else {
-      return Optional.empty();
-    }
+    initDuckdbSqlClient().ifPresent(client -> builder.put(DatabaseType.DUCKDB, client));
+    initSnowflakeClient().ifPresent(client -> builder.put(DatabaseType.SNOWFLAKE, client));
 
-    var url = (String) map.get("url");
-    if (Strings.isNullOrEmpty(url)) {
-      log.warn("Url must be specified in the snowflake engine");
-      return Optional.empty();
-    }
-    return Optional.of(url);
+    return builder.build();
   }
 
   @SneakyThrows
-  private SqlClient createSnowflakeClient(String url) {
-    Class.forName("net.snowflake.client.jdbc.SnowflakeDriver");
+  private Optional<SqlClient> initSnowflakeClient() {
+    var snowflakeConf = config.getSnowflakeConfig();
+    if (snowflakeConf == null) {
+      return Optional.empty();
+    }
 
-    var connectOptions =
-        new JDBCConnectOptions().setJdbcUrl(url + "?CLIENT_SESSION_KEEP_ALIVE=true");
+    var url = snowflakeConf.getUrl();
+    url += "?CLIENT_SESSION_KEEP_ALIVE=true";
+    var pool = initJdbcPool(snowflakeConf.getDriver(), url);
 
-    return JDBCPool.pool(vertx, connectOptions, new PoolOptions());
+    return Optional.of(pool);
   }
 
   @SneakyThrows
-  private SqlClient createDuckdbSqlClient() {
-    // In-memory DuckDB instance or you can specify a file path for persistence
-    var url = "jdbc:duckdb:";
-
-    Class.forName("org.duckdb.DuckDBDriver");
+  private Optional<SqlClient> initDuckdbSqlClient() {
+    var duckDbConf = config.getDuckDbConfig();
+    if (duckDbConf == null) {
+      return Optional.empty();
+    }
 
-    var connectOptions =
-        new JDBCConnectOptions().setJdbcUrl(url + "?" + DuckDBDriver.JDBC_STREAM_RESULTS + "=true");
+    var url = duckDbConf.getUrl();
+    var pool = initJdbcPool(duckDbConf.getDriver(), url);
 
-    return JDBCPool.pool(vertx, connectOptions, new PoolOptions());
+    return Optional.of(pool);
   }
 
-  private SqlClient createPostgresSqlClient() {
+  private SqlClient initPostgresSqlClient() {
     var poolOptions = new PoolOptions(this.config.getPoolOptions());
     // Note: setPipelined() method was removed in Vert.x 5, pipelining is now always enabled
     return Pool.pool(vertx, this.config.getPgConnectOptions(), poolOptions);
   }
+
+  @SneakyThrows
+  private Pool initJdbcPool(String driver, String url) {
+    Class.forName(driver);
```

Comment:
I wonder if we could turn this into a compile time check.

Instead of `initJdbcPool(duckDbConf.getDriver(), url);` have `initJdbcPool(org.duckdb.DuckDBDriver.class, url);`

Does it make sense to make jdbc driver configurable?!